spring-integration-webflux/src/main/java/org/springframework/integration/webflux/outbound/WebFluxRequestExecutingMessageHandler.java

```diff
@@ -169,13 +188,35 @@ protected Object exchange(Supplier<URI> uriSupplier, HttpMethod httpMethod, Http
 										ResponseEntity.status(response.statusCode())
 												.headers(response.headers().asHttpHeaders());
 
-								Mono<?> bodyMono = Mono.empty();
-
-								if (expectedResponseType instanceof ParameterizedTypeReference<?>) {
-									bodyMono = response.body(BodyExtractors.toMono((ParameterizedTypeReference<?>) expectedResponseType));
+								Mono<?> bodyMono;
+
+								if (expectedResponseType != null) {
+									if (this.replyToFlux) {
+										BodyExtractor<? extends Flux<?>, ReactiveHttpInputMessage> extractor;
+										if (expectedResponseType instanceof ParameterizedTypeReference<?>) {
+											extractor = BodyExtractors.toFlux(
+													(ParameterizedTypeReference<?>) expectedResponseType);
+										}
+										else {
+											extractor = BodyExtractors.toFlux((Class<?>) expectedResponseType);
+										}
+										Flux<?> flux = response.body(extractor);
+										bodyMono = Mono.just(flux);
+									}
+									else {
+										BodyExtractor<? extends Mono<?>, ReactiveHttpInputMessage> extractor;
+										if (expectedResponseType instanceof ParameterizedTypeReference<?>) {
+											extractor = BodyExtractors.toMono(
```

Comment:
Here, a additional check can be made to see if the wrapped type is `ClientResponse`, and if so, skip the `BodyExtractor`. I don't exactly remember how to get the type out of `ParameterizedTypeReference`, but I think Spring has some utilities for that.
Having the client provide a no-op `BodyExtractor` is not very convenient because they also have to be mindful of the order of the various `BodyExtractor`; and technically, we would not be extracting the body at all, so, a no-op `BodyExtractor` would be misleading.

Reply:
See my comment above.